### DISCUSSION TITLE
API is slow when deloy to Vercel

### DISCUSSION THREAD

**Discussion by meotimdihia** [vercel/next.js#22753] [Help] [answered]
I deployed this example to Vercel:
https://github.com/vercel/next.js/tree/canary/examples/api-routes-rest
This is deployed URL:
https://api-routes-rest-eight.vercel.app/

Screenshot: 
![chrome_2021-03-04_11-11-37](https://user-images.githubusercontent.com/300961/109910621-1f50c080-7cdb-11eb-8163-7ebfd00d329f.png)

It seems this API traveled cross 2 regions: `https://api-routes-rest-eight.vercel.app/`
`x-vercel-id: hkg1::sr2qf-1614831060863-502153a26f88`

Do we have a solution to this problem?

**Comment by meotimdihia** [comment]
It seems API is a serverless function and is deployed to a single region only.

**Comment by StefanSelfTaught** [comment, marked as answer]
The default region of Vercel serverless functions deployment is Washington DC, USA. The first request if for the HTML file which you get it from a Hong Kong CDN and then the serverless function.

**Comment by vennson** [comment]
@meotimdihia 
make sure ur Function Region is the same to where your DB/API server is. (this is in the settings of ur project in vercel.)

i had 1-2s before, but now it's averaging 150ms (faster and more consistent than in my local dev)
my problem was my supabase region is singapore, but my vercel function region is US by default

<img width="1245" alt="Screenshot 2024-04-16 at 11 57 35 AM" src="https://github.com/vercel/vercel/assets/28529262/f69d1ca8-4040-43dc-ba80-a708240a109c">
